include/ginkgo/core/base/executor.hpp

```diff
@@ -633,6 +633,18 @@ class Executor : public log::EnableLogging<Executor> {
      */
     virtual void synchronize() const = 0;
 
+    /**
+     * Verifies whether the executors share the same memory.
+     *
+     * @param other  the other Executor to compare against
+     *
+     * @return whether the executors this and other share the same memory.
+     */
+    bool memory_accessible(const std::shared_ptr<const Executor> &other) const
```

Comment:
Just curious: why do we take a reference to the `shared_ptr` instead of passing the `shared_ptr` by value? We don't want the pointer's internal reference count to be updated here?

Reply:
Yes, we don't need to take ownership here since we don't store the other executor anywhere etc, we only want to access it. Since Ginkgo isn't multithreaded (at least in terms of executor management), there is no issue such as the executor being destructed in between our access as well.